cmd/restic/cmd_forget.go

```diff
@@ -329,6 +329,26 @@ func addJSONSnapshots(js *[]Snapshot, list restic.Snapshots) {
 	}
 }
 
+// KeepReason helps to print KeepReasons as JSON with Snapshots with their ID included.
+type KeepReason struct {
+	Snapshot Snapshot `json:"snapshot"`
+	Matches  []string `json:"matches"`
+}
+
+func addJSONKeeps(js *[]KeepReason, list []restic.KeepReason) {
```

Comment:
Please change the function signature to `addJSONKeeps(list []restic.KeepReason) []KeepReason`. There's absolutely no need to use an output parameter here.

Reply:
With a focus on clean code and easy readability, I think it is more important to use the same solutions for the same requirements.
That's why I did it according to the following existing function:
func addJSONSnapshots(js *[]Snapshot, list restic.Snapshots)

Reply:
I changed for both functions the signature as suggested:

- asJSONSnapshots(list restic.Snapshots) []Snapshot
- asJSONKeeps(list []restic.KeepReason) []KeepReason

It is the first time for me doing a PR, so i have a Questions:
Should i squash the fixes with the first commit?

Reply:
I didn't notice that `addJSONSnapshots` was doing strange things :-) . Thanks for cleaning it up!

Yes please squash the commits (whether or not commits are squashed and when usually differs a lot between open source projects). Btw, I've that you've used to different names/emails in the commits. You might want to check which one should get merged.

Reply:
squash done and used correct e-mail (thanks for your notic)

Reply:
fork syncronized and commit rebased on head